### PR TITLE
[framework] add linked_table to moveos-stdlib

### PR DESCRIPTION
**Pull request by qShirley**
## Summary
I think this is a universal module and can be placed in moveos-stdlib.   I have added tests code and remove drop function, because `object::remove` should no field

### CODE REVIEW

File: frameworks/moveos-stdlib/sources/linked_table.move [hunk 1]
```diff
@@ -0,0 +1,230 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+/// Similar to `sui::table` but the values are linked together, allowing for ordered insertion and
```
Comment (jolestar) [inline frameworks/moveos-stdlib/sources/linked_table.move:4]: `moveos_std::table`

File: frameworks/moveos-stdlib/sources/linked_table.move [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    // Attempted to remove the front or back of an empty table
+    const ErrorTableIsEmpty: u64 = 1;
+
+    struct LinkedTable<K: copy + drop + store, phantom V: store> has key, store {
+        /// the number of key-value pairs in the table
+        size: u64,
+        /// the front of the table, i.e. the key of the first entry
+        head: Option<K>,
+        /// the back of the table, i.e. the key of the last entry
+        tail: Option<K>,
```
Comment (jolestar) [inline frameworks/moveos-stdlib/sources/linked_table.move:24]: We should put a `handle: Object<TablePlaceholder>` field in the LinkedTable and store the value using the handle Object.

File: frameworks/moveos-stdlib/sources/linked_table.move [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+    public fun new<K: copy + drop + store, V: store>(): Object<LinkedTable<K, V>> {
+        object::new(
+            LinkedTable {
+                size: 0,
+                head: option::none(),
+                tail: option::none(),
+            }
+        )
+
+    }
```
Comment (jolestar) [inline frameworks/moveos-stdlib/sources/linked_table.move:46]: The `new` fun should return LinkedTable<K,V>, not `Object<LinkedTable<K, V>>`

**Comment by jolestar** [comment]
cc @mx819812523

**Review by mx819812523 — APPROVED** [review]

**Comment by mx819812523** [comment]
LGTM👍 , Thanks for your contribution @qShirley

**Review by jolestar — APPROVED** [review]

**Pull request merged**